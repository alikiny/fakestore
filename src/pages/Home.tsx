import HomeBanner from "../components/home-components/HomeBanner"
import Collection from "../components/home-components/Collection"
import Box from "@mui/material/Box"

const Home = () => {
  return (
    <Box>
      <HomeBanner />
      <Collection />
    </Box>
  )
}

export default Home