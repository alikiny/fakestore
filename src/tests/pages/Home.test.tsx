import { customRender } from "../utils/render"

import Home from "../../pages/Home"

describe('Test Home page', () => {
    test('Should match page snapshot', () => {
        const component = customRender(<Home />)
        expect(component).toMatchSnapshot()
    })
    test('Should have Homebanner and Collections', () => {
        const { container } = customRender(<Home />)
        const collection = container.getElementsByClassName('home__collection')
        const homebanner = container.getElementsByClassName('banner')
        expect(collection.length).toEqual(1)
        expect(homebanner.length).toEqual(1)
    })
})