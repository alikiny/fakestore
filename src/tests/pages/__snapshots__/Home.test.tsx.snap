// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home page Should match page snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="banner MuiBox-root css-0"
          role="banner"
        >
          <div
            class="banner__body MuiBox-root css-0"
          >
            <h4>
              Only needed Ecommerce platform ever
            </h4>
            <h1>
              Beyond your expectations
            </h1>
            <hr />
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam eget turpis ut enim faucibus faucibus.
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 home__grid__container css-12n0uxm-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1equabv-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__grid__item css-bhp9pd-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <h3>
                    Realiability
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1equabv-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__grid__item css-bhp9pd-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <h3>
                    Quality
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1equabv-MuiGrid-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__grid__item css-bhp9pd-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <h3>
                    Speed
                  </h3>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="home__collection MuiBox-root css-0"
        >
          <h2>
            Browse the latest collection
          </h2>
          <div
            class="home__collection__grid MuiBox-root css-0"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="banner MuiBox-root css-0"
        role="banner"
      >
        <div
          class="banner__body MuiBox-root css-0"
        >
          <h4>
            Only needed Ecommerce platform ever
          </h4>
          <h1>
            Beyond your expectations
          </h1>
          <hr />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam eget turpis ut enim faucibus faucibus.
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 home__grid__container css-12n0uxm-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1equabv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__grid__item css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <h3>
                  Realiability
                </h3>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1equabv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__grid__item css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <h3>
                  Quality
                </h3>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1equabv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root home__grid__item css-bhp9pd-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <h3>
                  Speed
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="home__collection MuiBox-root css-0"
      >
        <h2>
          Browse the latest collection
        </h2>
        <div
          class="home__collection__grid MuiBox-root css-0"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
